Client/ClientGUI.py

```diff
@@ -0,0 +1,282 @@
+import json
+import socket
+import tkinter as tk
+import tkinter.font
+import urllib.request
+import uuid
+from queue import Queue, Full
+from threading import Event, Thread
+
+import pyaudio
+from ibm_cloud_sdk_core.authenticators import IAMAuthenticator
+from ibm_watson import SpeechToTextV1
+from ibm_watson.websocket import RecognizeCallback, AudioSource
+
+from EyeTracker import process_eye_tracking
+
+# PyAudio Configuration
+CHUNK = 1024
+BUF_MAX_SIZE = CHUNK * 10
+q = Queue(maxsize=int(round(BUF_MAX_SIZE / CHUNK)))
+FORMAT = pyaudio.paInt16
+CHANNELS = 1
+RATE = 44100
+
+# Create an instance of AudioSource
+audio_source = AudioSource(q, True, True)
+
+def get_uuid(mc_username):
+    global CLIENT_NAME
+    with urllib.request.urlopen(f'https://api.mojang.com/users/profiles/minecraft/{mc_username}') as response:
+        mc_profile = response.read().decode('utf-8')
+    try:
+        mc_username = json.loads(mc_profile)['name'] # ensures username is case-corrected
+        CLIENT_NAME = str(uuid.UUID(json.loads(mc_profile)['id']))
+        return f'Connected as {mc_username} ({CLIENT_NAME})'
+    except:
+        return 'Unable to retrieve UUID: invalid username or API is down'
+
+def connect_to_server(server_ip):
+    global CLIENT_SOCKET
+    HOST = server_ip
+    PORT = 5001
+    CLIENT_SOCKET = socket.socket()
+    try:
+        CLIENT_SOCKET.connect((HOST, PORT))
+        return f'Connected to {HOST}:{PORT}'
+    except:
+        return f'Unable to connect: IP may be incorrect'
+
+def connect_to_voice():
+    global SPEECH_TO_TEXT, CUSTOMIZATION_ID
+    with open('./client_credentials.json') as f:
+        credentials = json.load(f)
+        API_KEY = credentials['api_key']
+        SERVICE_URL = credentials['service_url']
+        CUSTOMIZATION_ID = credentials['customization_id']
```

Comment:
This could be modified to use actual values for api_key, service_url, and customization_id. Removes use of external json file for exe.

Reply:
Nevermind, you can just add the data file when compiling.